### DISCUSSION TITLE
flexibly align 2 molecules

### DISCUSSION THREAD

**Discussion by ccbiozhaw** [rdkit/rdkit#5413] [General]
I am struggling quite a bit with aligning 2 molecules.

What i have is a molecule (that may be corrupted, or just a substructure) and I want to superimpose the "correct" structure onto it. 
However, when i use the rdkit AlignMol method, it treats the mobile molecule as a rigid object. I want the mobile molecule to be able to flex so it can fit to the new structure best

AlignMol(target, mobile, atomMap = atomMap)

i would be very greatful for help

**Comment by ptosco** [comment]
@ccbiozhaw `AlignMol()` will only do a rigid body alignment, not a flexible one. What you could is pre-generate a number of conformers, align each of them on the template molecule with `AlignMol()` and choose the conformer that yields the best alignment (e.g., in terms of RMS deviation).

**Reply by ccbiozhaw** [reply]
hmm, I need the alignment to be as close as possible. What I currently do is match all the atoms I can (based on their atom type and neighborhood), then overwrite the coordinates. Then I can fix/correct the atoms that couldn't be matched by doing a geometry optimization, fixing the matched atoms in place. 


this works fine in a decent amount of cases, however very tedious to adjust in the cases it doesnt.

**Reply by ptosco** [reply]
You might combine the two approaches, i.e. start from the lowest RMSD conformation and copy coordinates across, then minimise. That would reduce odds of chirality inversions and other undesired artefacts.

**Reply by MichelML** [reply]
@ccbiozhaw you might want to have a look at this function in datamol https://github.com/datamol-org/datamol/blob/main/datamol/align.py#L35 , which uses a MCS approach if there is no clear substruct match between the two molecules.

you might want to try to install datamol and use the function as is to see if it fits your need too
